02_cleaning.Rmd

```diff
@@ -1035,3 +1035,212 @@ print(objSDC, "risk")
+        prog <- 1
+        for (temp_resp_batch in temp_resp_batches){
+          tcltk::setTkProgressBar(pb, prog, progress.bar.title, paste0("Number of batches executed: ", prog, " of ", batching,"\n",length(temp_resp_batch$input_vec)," responses will be translated from ",code," to ",target_lang, "\nThis means ",sum(str_length(temp_resp_batch$input_vec))," utf-8 characters."))
+          prog <- prog + 1
+          # actual translation:
+          result_vec <- NULL
+          result_vec <- try(translateR::translate(content.vec = temp_resp_batch$input_vec,
+                              microsoft.api.key = source("resources/microsoft.api.key_ukraine.R")$value,
+                              microsoft.api.region = "switzerlandnorth",
+                              source.lang = code, target.lang = target_lang))
+          if(inherits(result_vec,"try-error")) break
+          # checking the results
+          info_df <- rbind(info_df, data.frame(## DEBUGG IT HERE
+            "input_responses_num" = length(temp_resp_batch$input_vec),
+            "translated_characters_num" = sum(str_length(temp_resp_batch$input_vec)),
+            "language_from" = code,
+            "result_num" = length(result_vec),
+            "time_elapsed" = as.numeric(Sys.time() - start_time),
+            "date"=Sys.Date(),
+            "status"=NA))
+          if(is.null(result_vec)){
+            warning("Error while translating responses: result_vec is NULL\n")
+            info_df$status <- "error"
+          }else{
+            temp_resp_batch[[col_name]] <- gsub("&#39;", "'", result_vec)
+            if(length(result_vec) == length(temp_resp_batch$input_vec)){ 
+              info_df$status <- "success"
+              # bind the translated and source dfs
+              temp_resp_whole <- rbind(temp_resp_whole,temp_resp_batch)
+            }else{
+              info_df$status <- "partial success"
+            }
+          }
+        }
+        close(pb)
+        if("partial success" %in% info_df$status){
+          svDialogs::msgBox("translate.responses: finished - PARTIAL SUCCESS?")
+        } else{
+          svDialogs::msgBox("translate.responses: finished - SUCCESS")
+        }
+        responses <- responses %>% left_join(temp_resp_whole, by = c("resp_lower" = "input_vec")) 
+      }
+    }else{
+      warning("Nothing to be translated")
+    }
+  }
+  # dump info about the results of translation
+  log_filename <- "translate_info.csv"
+  if(file.exists(log_filename)) write.table(info_df, file = log_filename, append = T, row.names = F, col.names = F, sep = ',')
+  else write.table(info_df, file = log_filename, row.names = F, col.names = T, sep = ',')
+
+  responses <- responses %>% select(-resp_lower)
+  return(responses)
+}
+```
+
+3. Example:
+
+```{r}
+other_columns <- c("reason_bad_relationship_other")
+  tool.survey %>% filter(type == "text") %>% pull(name)
+
+responses <- find.responses(data.list$main,other_columns,values_to = "response.uk")
```

Comment:
Could you please check the example here to be using main dataset and kobo tool included in the Rcookbook folder? To make it easy for anyone using this code. Thanks :)